### PR TITLE
DAOS-11485 object, tests: fixed several coverity issues

### PR DESCRIPTION
**Pull request by wiliamhuang**
119034 Use of 32-bit time_t
128956 Unchecked return value
449349 Unchecked return value
21893 Unchecked return value
449350 Derefence after null check

Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — CHANGES_REQUESTED** [review]
Is this supposed to be a 2.2 cherry-pick?

**Comment by wiliamhuang** [comment]
> Is this supposed to be a 2.2 cherry-pick?

I created master PR first and got many CI issues due to regression previously. Later I mainly worked on 2.2 PR. "unclean cherry-pick" may be not a valid label. Just removed it. Thank you!

**Review by daltonbohning — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@jolivier23 Could you please review? Thank you!

File: src/object/srv_obj_migrate.c [hunk 1]
```diff
@@ -189,8 +189,11 @@ obj_tree_create(daos_handle_t toh, void *key, size_t key_size,
 
 	*rootp = tmp_root;
 out:
-	if (rc < 0)
-		dbtree_delete(toh, BTR_PROBE_EQ, &key_iov, NULL);
+	if (rc < 0) {
+		rc2 = dbtree_delete(toh, BTR_PROBE_EQ, &key_iov, NULL);
+		if (rc2)
+			D_WARN("failed to delete \"%s\": "DF_RC"\n", (char *)key, DP_RC(rc2));
```
Comment (jolivier23) [inline src/object/srv_obj_migrate.c:195]: This is not correct.  It needs to be DF_KEY and DP_KEY.  This has a couple of effects

1. On release builds, it doesn't print any data (which may be private)
2. It only prints a portion of a long key
3. It prints an integer key as an integer
4. It doesn't print non-ascii characters

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/object/srv_obj_migrate.c (hunk 1 above)
Comment (wiliamhuang) [reply src/object/srv_obj_migrate.c]: I just changed it to, 
D_WARN("failed to delete "DF_KEY": "DF_RC"\n", DP_KEY(&key_iov), DP_RC(rc2));

Thank you very much!

**Review by wangshilong — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@jolivier23 Could you please review and land? Thank you!

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**